### PR TITLE
18123 rebased legal name FB with main

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* /bcgov/entity#18123

*Description of changes:*
Rebased feature branch with main

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Dashboard/FilingHistoryList/FilingTemplate.vue [hunk 1]
```diff
@@ -53,7 +53,10 @@
       </div>
     </v-expansion-panel-header>
 
-    <v-expansion-panel-content class="pt-5">
+    <v-expansion-panel-content
+      eager
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/FilingTemplate.vue:57]: General note: we should probably use `eager` for all expansion panel content blocks to facilitate unit testing.

File: src/views/Dashboard.vue [hunk 2]
```diff
@@ -300,6 +300,7 @@ export default {
         'isAdminFrozen',
         'isBenBcCccUlc',
         'isFirm',
+        'isFutureEffectiveAmalgamation',
```
Comment (severinbeauvais) [inline src/views/Dashboard.vue:303]: Duplicate reference. To keep things sorted, please delete line 299.

File: tests/unit/FilingHistoryList2.spec.ts [hunk 3]
```diff
@@ -145,6 +150,44 @@ filings.forEach((filing: any, index: number) => {
...
+      expect(vm.getFilings.length).toBe(1) // sanity check
+      const item = vm.getFilings[0]
+      expect(item.businessIdentifier).toBe('BC1234567')
+      expect(item.data.amalgamationApplication.type).toBe(FilingSubTypes.AMALGAMATION_HORIZONTAL)
+      expect(item.displayName).toBe('Amalgamation Application - Horizontal')
+      expect(item.name).toBe(FilingTypes.AMALGAMATION_APPLICATION)
+      expect(item.status).toBe(FilingStatus.COMPLETED)
+
+      // verify display
+      expect(wrapper.find('.item-header-title').text()).toBe('Amalgamation Application - Horizontal')
```
Comment (severinbeauvais) [inline tests/unit/FilingHistoryList2.spec.ts:164]: This is an old title (has been changed recently). Probably the displayName in the JSON file needs to be updated.

File: src/components/Dashboard/FilingHistoryList/FilingTemplate.vue (hunk 1 above)
Comment (JazzarKarim) [reply src/components/Dashboard/FilingHistoryList/FilingTemplate.vue:57]: Noted.

File: src/views/Dashboard.vue (hunk 2 above)
Comment (JazzarKarim) [reply src/views/Dashboard.vue:302]: Thanks for catching this Sev.

File: tests/unit/FilingHistoryList2.spec.ts (hunk 3 above)
Comment (JazzarKarim) [reply tests/unit/FilingHistoryList2.spec.ts:164]: I'm looking at main and I see the line is exactly the same Sev:
https://github.com/bcgov/business-filings-ui/blob/43b527cce8469b8df1a22f562799ae740f568cd7/tests/unit/FilingHistoryList2.spec.ts#L164

Do we want it to be something different?

Comment (severinbeauvais) [reply tests/unit/FilingHistoryList2.spec.ts:164]: Yeah, these filings have new titles - see https://github.com/bcgov/lear/pull/2469.

Unfortunately, unit tests are happy with bad data if that's what you're comparing to 😆

Comment (severinbeauvais) [reply tests/unit/FilingHistoryList2.spec.ts]: Much appreciated! In the end, it doesn't matter if this is fixed in main or LE f/b.

**Review by severinbeauvais — APPROVED** [review]

**Review by argush3 — APPROVED** [review]

**Pull request merged**